test/e2e/main_test.go

```diff
@@ -9,25 +9,34 @@ import (
 	"rhobs/monitoring-stack-operator/test/e2e/framework"
 	"testing"
 
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
 	ctrl "sigs.k8s.io/controller-runtime"
 	"sigs.k8s.io/controller-runtime/pkg/log/zap"
 )
 
 var (
 	f *framework.Framework
 
-	// TODO(fpetkovski): change once we are able to deploy the operator to a different namespace
-	e2eTestNamespace = "default"
+	e2eTestNamespace = "e2e-tests"
 )
 
 func TestMain(m *testing.M) {
+	main(m)
```

Comment:
Deferred calls are not executed on `os.Exit` from TestMain. As a workaround, we call another function in which we can add deferred calls.

Reply:
Good idea